### PR TITLE
Adds TensorFlow mixin

### PR DESCRIPTION
**Pull request by StefanKurek**

### CODE REVIEW

File: tensorflow-mixin/dashboards/tensorflow-overview.libsonnet [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+    tooltip: {
+      mode: 'single',
+      sort: 'none',
+    },
+  },
+  targets: [
+    prometheus.target(
+      'rate(:tensorflow:serving:request_count{job=~"$job",instance=~"$instance",model_name=~"$model_name"}[$__rate_interval])',
+      datasource=promDatasource,
+    ),
```
Comment (BinaryFissionGames) [inline tensorflow-mixin/dashboards/tensorflow-overview.libsonnet:86]: For all these prometheus queries, I'd recommend to add a legend format (e.g.: https://github.com/grafana/jsonnet-libs/blob/fd5379a1fba2d572fc314a0395dd61e7df335948/gitlab-mixin/dashboards/gitlab-overview.libsonnet#L225) - This removes a lot of irrelevant or duplicated info for the dashboard panel.

File: tensorflow-mixin/dashboards/tensorflow-overview.libsonnet [hunk 2]
```diff
@@ -0,0 +1,740 @@
+local g = (import 'grafana-builder/grafana.libsonnet');
+local grafana = (import 'grafonnet/grafana.libsonnet');
+local dashboard = grafana.dashboard;
+local template = grafana.template;
+local prometheus = grafana.prometheus;
+
+local dashboardUid = 'tensorflow-overview';
+local matcher = 'job=~"$job",instance=~"$instance",model_name=~"$model_name"';
```
[the remark this answers is not in the document]
Comment (StefanKurek) [reply tensorflow-mixin/dashboards/tensorflow-overview.libsonnet:8]: good catch, i forgot to remove that

**Review by BinaryFissionGames — APPROVED** [review]

File: tensorflow-mixin/alerts/alerts.libsonnet [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                 'which is above the threshold %(alertsModelRequestErrorRate)s%%, ' +
+                 'indicating a potentially larger issue for {{$labels.instance}}') % $._config,
+            },
+          },
+          {
+            alert: 'TensorFlowHighBatchQueuingLatency',
+            expr: |||
+              increase(:tensorflow:serving:batching_session:queuing_latency_sum[2m]) / increase(:tensorflow:serving:batching_session:queuing_latency_count[2m]) > %(alertsBatchQueuingLatency)s
+            ||| % $._config,
+            'for': '1m',
```
Comment (gaantunes) [inline tensorflow-mixin/alerts/alerts.libsonnet:29]: Is it necessary to be 1m? Could it follow the 5m rule?

File: tensorflow-mixin/config.libsonnet [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  _config+:: {
+    dashboardTags: ['tensorflow-mixin'],
+    dashboardPeriod: 'now-1h',
```
Comment (gaantunes) [inline tensorflow-mixin/config.libsonnet:4]: We normally apply a 30min period as standard to our dashboards.

File: tensorflow-mixin/alerts/alerts.libsonnet (hunk 3 above)
Comment (StefanKurek) [reply tensorflow-mixin/alerts/alerts.libsonnet:29]: @gaantunes I can definitely make this change. The spec didn't call it out so I wasn't sure.

File: tensorflow-mixin/config.libsonnet (hunk 4 above)
Comment (StefanKurek) [reply tensorflow-mixin/config.libsonnet:4]: I can make this change.

File: tensorflow-mixin/alerts/alerts.libsonnet (hunk 3 above)
Comment (StefanKurek) [reply tensorflow-mixin/alerts/alerts.libsonnet]: @gaantunes updated

File: tensorflow-mixin/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+| TensorFlowModelRequestHighErrorRate | The TensorFlow requests for a model have a large percentage of failures. Could be indicative that the model for the instance is experiencing serious problems. |
+| TensorFlowHighBatchQueuingLatency | The TensorFlow queued batches are spending a long time in the queue. Could be indicative of problems with the batch queuing for the instance. |
+
+## Generating dashboards and alerts
+
+```bash
+make
+```
+
+Creates a generated `dashboards_out` directory and `prometheus_alerts.yaml` that can be imported into Grafana.
```
[the remark this answers is not in the document]
Comment (StefanKurek) [reply tensorflow-mixin/README.md]: @gaantunes updated

File: tensorflow-mixin/config.libsonnet (hunk 4 above)
Comment (StefanKurek) [reply tensorflow-mixin/config.libsonnet]: @gaantunes updated

**Review by gaantunes — APPROVED** [review]
LGTM, thanks for the changes.

**Pull request merged**